Windows fixes

rivy opened the pull request:
I wanted to give a little back to your project as a way of saying thanks for addressing my earlier issue (#97) so quickly.

So, I polished the windows story a bit.

Initially, I was just going to supply a working windows build recipe for AppVeyor CI, but I got caught up in getting all CI tests working...

1. I updated the AppVeyor config to a version which addresses both #82 and #96. The 'gnu' ABI build failures are addressed with a simple work-around. The work-around is marked in the AppVeyor config file and includes references to the rust issues addressing the problem.

2. I had to pin the 'cc' crate in onig_sys to v1.0.18 to successfully create a windows/rust v1.27.0 build. Later patch versions of 'cc' (specifically, v1.0.29) use `trim_start_matches` which causes an error for windows/rust v1.27.0 compilation.

3. The i686 test segfault is addressed by isolating the two test lines which cause the segfaults. I don't have a deep enough understanding of the code to go any further. But it points to the problem code, commenting it out, and should give you a direction to pursue further remediation.

4. The Travis CI config is relaxed and now tests non-'master' branches (to CI test my feature branch).

The code is separated into commits along the lines noted above. Notably, the test segfault correction commit is included to get clean AppVeyor CI builds and as a pointer to the problem, not a solution. So, it'll need work to actually solve the issue.

* [AppVeyor CI](https://ci.appveyor.com/project/rivy/rust-onig/builds/22268318) and [Travis CI](https://travis-ci.org/rivy/rust.onig/builds/491465824)

But, anyway, I'm glad to re-work them as you direct. Just let me know...

iwillspeak commented on onig/src/names.rs:
```diff
@@ -169,13 +169,13 @@ mod tests {
     #[test]
     fn test_regex_names() {
```
Could you mark this test as `#[ignore]` rather than commenting out parts of it. I've created an issue (#100) for you to link to in a FIXME comment too.

iwillspeak commented on .appveyor.yml:
```diff
@@ -0,0 +1,159 @@
+    FEATURES: ""
+  matrix:
+  # minimum version
+  - CHANNEL: 1.27.0
```
Nice! Good to be testing with 1.27 on Windows too.

iwillspeak commented on .travis.yml:
```diff
@@ -10,9 +10,10 @@ rust:
-branches:
-  only:
-    - master
+# branches:
```
Should we match the Appveyor config and exclude `gh-pages` here? Also I _think_ this was origionally to stop us getting two builds for pull requests. It doesn't seem to be needed any more though.

iwillspeak commented on onig_sys/Cargo.toml:
```diff
@@ -25,7 +25,7 @@ license = "MIT"
 [build-dependencies]
 pkg-config = "0.3"
-cc = "1.0"
+cc = "=1.0.18"
```
Maybe a comment linking to this PR to explain that this is pinned for a reason.

rivy commented:
Thanks for the suggestions.

I made several improvements based on your feedback.

* I gated the segfaulting test to be ignored specifically only on windows/i686 builds.
* I added several FIXME comments (with some explanation and issue references) as bookmarks for later fixes.
* I added 'gh-pages' as an 'except' branch for Travis CI.

I'm just waiting on successful CI completion, then I'll force push back up to this PR.

If it needs any more polish, let me know.

iwillspeak commented:
Super work! Merged 🚀
